### PR TITLE
[MPQEditor] Implement 'createMPQJson'

### PR DESCRIPTION
**Pull request by stamalakhov**
This commit implements 'createMPQJson' for creation/editing default mpq file.

I've checked its validity in the draft #1511. Please see https://github.com/Samsung/ONE-vscode/pull/1511#issue-1642481493 for its correctness.

Fresh draft: https://github.com/Samsung/ONE-vscode/pull/1511
Previous draft: https://github.com/Samsung/ONE-vscode/pull/1505
Related: https://github.com/Samsung/ONE-vscode/issues/1491

ONE-vscode-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

**Review by seanshpark — APPROVED** [review]
LGTM

File: src/MPQEditor/MPQEditor.ts [hunk 1]
```diff
@@ -83,6 +84,61 @@ export class MPQEditorProvider implements vscode.CustomTextEditorProvider {
...
+      })
+      .then((value) => {
+        if (!value) {
+          Logger.debug("MPQEditor", "User hit the escape key!");
+          return;
+        }
+
+        MPQEditorProvider.createDefaultMPQ(value!, dirPath, circleName).then(
+          (uri) => {
+            if (uri !== undefined) {
```
Comment (dayo09) [inline src/MPQEditor/MPQEditor.ts:124]:
```suggestion
            if (uri) {
```

(optional) This is simpler :-D

File: src/MPQEditor/MPQEditor.ts [hunk 2]
```diff
@@ -83,6 +84,61 @@ export class MPQEditorProvider implements vscode.CustomTextEditorProvider {
...
+              vscode.commands.executeCommand(
+                "vscode.openWith",
+                uri,
+                MPQEditorProvider.viewType
+              );
+            } else {
+              Logger.error(
+                "MPQEditor",
+                "createMPQJson",
+                `Failed to create mpq file for the ${value!}`
```
Comment (dayo09) [inline src/MPQEditor/MPQEditor.ts:134]:
```suggestion
                `Failed to create mpq file for the ${value}!`
```

Comment (stamalakhov) [reply src/MPQEditor/MPQEditor.ts:134]: Ahh. Sorry.

**Review by dayo09 — APPROVED** [review]
LGTM, Thanks :-D

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**